File: backend/app/schemas/member.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class MemberBase(BaseModel):
    member_name: str
    position: str
    tech_stack: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None


class MemberCreate(MemberBase):
    pass


class MemberUpdate(BaseModel):
    member_name: Optional[str] = None
    position: Optional[str] = None
    tech_stack: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None


class MemberBrief(BaseModel):
    member_id: int
    member_name: str
    position: str

    model_config = ConfigDict(from_attributes=True)


class Member(MemberBase):
    member_id: int
    create_time: datetime
    update_time: datetime

    model_config = ConfigDict(from_attributes=True)


class MemberPage(BaseModel):
    items: list[Member] = Field(default_factory=list)
    total: int
    skip: int = 0
    limit: int

File: frontend/src/types/member.ts
import type { PaginatedResponse } from "@/types/common";

export interface MemberBrief {
  member_id: number;
  member_name: string;
  position: string;
}

export interface Member extends MemberBrief {
  tech_stack: string | null;
  phone: string | null;
  email: string | null;
  create_time: string;
  update_time: string;
}

export interface MemberLeadProjectSummary {
  project_id: number;
  project_name: string;
  project_type: string;
  project_status: string;
  business_unit: string;
  update_time: string;
}

export interface MemberResourceProjectSummary {
  project_id: number;
  project_name: string;
  project_type: string;
  project_status: string;
  business_unit: string;
}

export interface MemberDevelopedResourceSummary {
  resource_id: number;
  resource_type: string;
  git_repo: string | null;
  tech_framework: string | null;
  deploy_branch: string | null;
  prod_domain: string | null;
  update_time: string;
  project: MemberResourceProjectSummary | null;
}

export interface MemberDetail extends Member {
  lead_projects: MemberLeadProjectSummary[];
  developed_resources: MemberDevelopedResourceSummary[];
}

export interface MemberPayload {
  member_name: string;
  position: string;
  tech_stack?: string | null;
  phone?: string | null;
  email?: string | null;
}

export interface MemberQueryParams {
  skip?: number;
  limit?: number;
  search?: string;
}

export type MemberPage = PaginatedResponse<Member>;
